### ISSUE TITLE
should provide easy convert function from color to vector?

### ISSUE THREAD

**Issue by babaq** [JuliaGraphics/ColorTypes.jl#119]
we could easily convert a `Vector` to `RGB`/`RGBA` etc. but it doesn't allow to just `Vector(RGB/RGBA)`, or `vec(RGB/RGBA)`, it could be done by `reducec(vcat,[],RGB/RGBA)`, but should it be added as a straitforward converting functions?

**Comment by timholy** [comment]
You can call the constructor directly:

```julia
julia> using ColorTypes

julia> rgbvec = rand(3)
3-element Array{Float64,1}:
 0.12561714622439624
 0.29186235848361486
 0.7310749506094338 

julia> RGB(rgbvec...)
RGB{Float64}(0.12561714622439624,0.29186235848361486,0.7310749506094338)
```

Are you also asking about the reverse transformation? If you want a single channel value, `red(c)` is typically far clearer than `cvec[1]`, and is robust to storage-order issues (we have not just RGB but BGR and RGB1 and RGB4 and perhaps soon [RGB565](https://github.com/JuliaGraphics/ColorTypes.jl/issues/114), etc).

Also note that ImageCore.jl has `colorview`, `channelview`, and `reinterpretc` if you are working with arrays. See the [JuliaImages documentation](https://juliaimages.org/latest/) for more info.

There's a subtle difference between `convert` and construction that makes me uneasy about calling this `convert(RGB, rand(3))`. For example, if you have a `Meters` type then you should be able to say `Meters(1.2)` just fine, but `convert(Meters, 1.2)` is worrisome because dimensionless 1.2 is not the same thing as 1.2m. If you allowed `convert`, then you could do this: `convert(Seconds, convert(Float64, Meters(1.2))` and suddenly you've asserted that 1.2s == 1.2m, which is rather disastrous. The analog here is `convert(RGB, convert(Vector, BGR(1, 0.3, 0.8)))` which will surprise *someone* no matter how you think these operations should work.

**Comment by babaq** [comment]
Yes the name `convert` may be a little confusing, The constructor will be clearer. but in reverse direction, the Colorant does not support iterators, such as `vec = [RGB...]`, i noticed it's been discussed in an old issue #7 , and i think it will be very helpful, since Matrix multiplication `M*color = convertedcolor` sometimes is natrual to convert color from one space to another.

Converting from a vector to a color means we are aware the color type constructor argument order, so does the reverse: when convert a color to a vector we usually mean the same constructor argument order. the `Images.channelview` does it, but it only accept array of colors, not signal color value.

there are risks of losing sematics of color when converting to vector, and I think it's user's responsibility to know what they are doing, and of course it should explicitly documented the risk of such conversion.

**Comment by timholy** [comment]
I do see the point. If you want to submit a PR, I'd consider it. Presumably one would want to use a `StaticVector`, but there would be negatives to making this small package depend on StaticArrays. Maybe via Requires?

**Comment by babaq** [comment]
I don't know if `StaticVector` is a good idea, but sometimes you want to put all the colors in 3xN/Nx3 and multiply with the conversion Matrix, use plain `Vector` might be easier to manipulate them.

**Comment by timholy** [comment]
Why can't you use `channelview` for that case?

**Comment by babaq** [comment]
yes, `channelview` will do in that case
